jetty9/src/main/java/com/linecorp/armeria/server/jetty/ArmeriaEndPoint.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.jetty;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+
+import javax.annotation.Nullable;
+
+import org.eclipse.jetty.io.Connection;
+import org.eclipse.jetty.io.EndPoint;
+import org.eclipse.jetty.io.FillInterest;
+import org.eclipse.jetty.io.WriteFlusher;
+import org.eclipse.jetty.util.Callback;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+final class ArmeriaEndPoint implements EndPoint {
+
+    private static final AtomicReferenceFieldUpdater<ArmeriaEndPoint, State> stateUpdater =
+            AtomicReferenceFieldUpdater.newUpdater(ArmeriaEndPoint.class, State.class, "state");
+
+    private final ServiceRequestContext ctx;
+    @Nullable
+    private final String hostname;
+    @Nullable
+    private volatile InetSocketAddress localAddress;
+    @Nullable
+    private volatile Connection connection;
+    private volatile State state = State.OPEN;
+
+    // Helpers
+    private final FillInterest fillInterest = new FillInterest() {
+        @Override
+        protected void needsFillInterest() {}
+    };
+    private final WriteFlusher writeFlusher = new WriteFlusher(this) {
+        @Override
+        protected void onIncompleteFlush() {}
+    };
+
+    ArmeriaEndPoint(ServiceRequestContext ctx, @Nullable String hostname) {
```

Comment:
It seems like `hostname` is always not null.

Reply:
Changed the logic a little bit so that the final hostname is determined when `ArmeriaEndPoint` is instantiated. `hostname` remains nullable therefore.

Reply:
Ah, I was talking about when the `ArmeriaEndPoint` is created in `JettyService`. 😅 
The `hostname` is set in `serviceAdded()` and `ArmeriaEndPoint` is created in `servce` method,
so I thought it is not null.
But I'm fine with the logic now though. 😄